src/java.base/share/classes/java/util/stream/Gatherer.java

```diff
@@ -0,0 +1,571 @@
+     * @param <T> the type of input elements for the new gatherer
+     * @param <A> the type of initializer for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, A, R> Gatherer<T, A, R> ofSequential(
+            Supplier<A> initializer,
+            Integrator<A, T, R> integrator) {
+        return of(
+                initializer,
+                integrator,
+                defaultCombiner(),
+                defaultFinisher()
+        );
+    }
+
+    /**
+     * Returns a new, sequential, {@code Gatherer} described by the given
+     * {@code initializer}, {@code integrator}, and {@code finisher}.
+     *
+     * @param initializer the supplier function for the new gatherer
+     * @param integrator the integrator function for the new gatherer
+     * @param finisher the finisher function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <A> the type of initializer for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, A, R> Gatherer<T, A, R> ofSequential(
+            Supplier<A> initializer,
+            Integrator<A, T, R> integrator,
+            BiConsumer<A, Downstream<? super R>> finisher) {
+        return of(
+                initializer,
+                integrator,
+                defaultCombiner(),
+                finisher
+        );
+    }
+
+    /**
+     * Returns a new, parallelizable, and stateless {@code Gatherer} described
+     * by the given {@code integrator}.
+     *
+     * @param integrator the integrator function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, R> Gatherer<T, Void, R> of(Integrator<Void, T, R> integrator) {
+        return of(
+                defaultInitializer(),
+                integrator,
+                Gatherers.Value.DEFAULT.statelessCombiner,
+                defaultFinisher()
+        );
+    }
+
+    /**
+     * Returns a new, parallelizable, and stateless {@code Gatherer} described
+     * by the given {@code integrator} and {@code finisher}.
+     *
+     * @param integrator the integrator function for the new gatherer
+     * @param finisher the finisher function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, R> Gatherer<T, Void, R> of(
```

Comment:
Nit: In `Collector` the parameters are formatted in a different way. Should we be consistent with that existing formatting?

Reply:
Which parameters? The type parameters or the method parameters?

Reply:
Here is an example:

```
public static<T, A, R> Collector<T, A, R> of(Supplier<A> supplier,
                                                 BiConsumer<A, T> accumulator,
                                                 BinaryOperator<A> combiner,
                                                 Function<A, R> finisher,
                                                 Characteristics... characteristics) {
```

Reply:
I'd love to, but if I want to keep horizontal space manageable I'll have to break somewhere..